### ISSUE TITLE
EG does not cleanup Envoy proxy deployments if a gateway delete notification is missed

### ISSUE THREAD

**Issue by alexwo** [envoyproxy/gateway#4792]
*Description*:
The issue observed is that orphaned Envoy proxy deployments are not removed when a gateway delete notification is missed. It is expected that during a full reconcile, EG can identify and clean up these orphaned proxies to maintain consistency and prevent resource leakage.

*Repro steps*:
	1.	Create a gateway and observe that the Envoy proxy deployment is created successfully.
	2.	Scale EG to 0 and delete the gateway resource
	3.	Scale EG back to 1 to force a full reconcile
	4.	Observe that the orphaned Envoy proxy deployment is not removed.

**Comment by arkodg** [comment]
this can be solved with https://github.com/envoyproxy/gateway/issues/3051

**Comment by Skaronator** [comment]
Same with PDB and HPA resources. They won't get deleted when removing a Gateway or disabling them in the EnvoyProxy config

**Comment by arkodg** [comment]
#3051 cannot be implemented because ownerReferences dont have cross namespace support

**Comment by MaYuan-02** [comment]
> Same with PDB and HPA resources. They won't get deleted when removing a Gateway or disabling them in the EnvoyProxy config

@arkodg  I'm also encountering issues with deleting HPA and PDB resources. Is anyone currently working on resolving this? If not, I'd be happy to take a stab at it.

**Comment by arkodg** [comment]
hey @MaYuan-02 are you hitting https://github.com/envoyproxy/gateway/issues/4380 ?

**Comment by arkodg** [comment]
reg this issue, we may need to implement mark and sweep in the infra init function, so it cleans up orphaned resources that once belonged to it

**Comment by MaYuan-02** [comment]
@arkodg Thanks for your reply.
I'm facing an issue where turning off HPA in the EnvoyProxy resource doesn't remove the HPA resource, leading to mismatches with the gateway's replica count. The control plane only deletes resources when it receives an InfraIR delete message, which disabling HPA doesn't trigger. Should I open a new issue to address this?

**Comment by MaYuan-02** [comment]
> reg this issue, we may need to implement mark and sweep in the infra init function, so it cleans up orphaned resources that once belonged to it
You're right, we could add logic to clean up orphaned resources in the initInfra after the leader election. Is anyone currently working on fixing this issue? If not, I'd be happy to give it a try.

**Comment by cabrinha** [comment]
> > reg this issue, we may need to implement mark and sweep in the infra init function, so it cleans up orphaned resources that once belonged to it
> 
> Yes, we could add logic to clean up orphaned resources in the initInfra after the leader election. Is anyone currently working on fixing this issue? If not, I'd be happy to give it a try.

Go ahead and give it a try.

Were ownerRef's ever fixed? I'm deploying gateway-helm and it's all in the same namespace, but ArgoCD still doesn't show my GatewayClass resources. Everything is deployed to the `envoyproxy-gateway-system` namespace.

Here is my OwnerRef from the `Deployment` that I cannot see in ArgoCD:
```yaml
  ownerReferences:
  - apiVersion: gateway.networking.k8s.io/v1
    kind: GatewayClass
    name: envoy
    uid: 9daec19a-0b2f-42e3-88e8-72b0a02ba3ab
```

How can I make this show up on the ArgoCD UI?

**Comment by arkodg** [comment]
GatewayClass is cluster scoped, which maybe why its not showing it
